File: src/main/java/liqp/filters/Downcase.java
package liqp.filters;

import liqp.nodes.RenderContext;

class Downcase extends Filter {

    /*
     * downcase(input)
     *
     * convert a input string to DOWNCASE
     */
    @Override
    public Object apply(RenderContext context, Object value, Object... params) {

        return super.asString(value).toLowerCase();
    }
}

File: liqp-core/src/main/kotlin/liqp/filter/Downcase.kt
package liqp.filter

import liqp.context.LContext
import liqp.params.FilterParams

class Downcase : LFilter() {

  /**
   * downcase(input)
   *
   * convert a input string to DOWNCASE
   */
  override fun onFilterAction(context: LContext, value: Any?, params: FilterParams): Any? {
    return context.asString(value)?.toLowerCase()
  }
}
